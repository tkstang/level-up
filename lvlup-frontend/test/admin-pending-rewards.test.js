import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';
import PendingRewards from '../src/components/Admin/dashboard/pending-requests';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('PendingRewards should render', () => {
  const store = mockStore({ lvlupApp });
  const approveReward = shallow(
    <PendingRewards pendingRequests={{ requestsAdmin: { length: 5 } }} store={store} />,
  );
  expect(shallowToJson(approveReward)).toMatchSnapshot();
});
