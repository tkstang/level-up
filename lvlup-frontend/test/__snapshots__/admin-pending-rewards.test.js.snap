exports[`test PendingRewards should render 1`] = `
<PendingRequestsTable
  approveSelectedReward={[Function]}
  denySelectedReward={[Function]}
  requestsAction={[Function]}
  resetPendingRequests={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  } />
`;
